sdk/state/state.go

```diff
@@ -36,6 +36,40 @@ func NewChannel(c Config) *Channel {
 	return channel
 }
 
+type Snapshot struct {
```

Comment:
💭  thoughts on moving the Snapshot structs and logic to a snapshot.go file? It would make the state.go file less crowded and easier to tie the snapshot_test.go tests to the code it's testing

Reply:
I think we should keep the `New` functions and their related types close to each other. That's a good point with the disconnect with `snapshot_test.go` though. I renamed it to `state_test.go`.